### PR TITLE
Add icon to catalog category

### LINKED ISSUE #2605: CatalogCategory should have icon
Icon should support both material and custom svg icons.

UI mockup:
image

### PR DESCRIPTION
**Pull request by jakolehm**
Implements initial support for `CatalogCategory` icons. Currently icon is only shown in hotbar but we should show this also in Catalog (out-of-scope for this PR).

Also improves `Icon` component to support custom svgs.

image

Fixes #2605

### CODE REVIEW

**Review by Nachasic — APPROVED** [review]
Looks good

File: src/renderer/components/icon/icon.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ export class Icon extends React.PureComponent<IconProps> {
 
     // render as inline svg-icon
     if (svg) {
-      const svgIconText = require(`!!raw-loader!./${svg}.svg`).default;
+      const svgIconText = svg.includes("<svg") ? svg : require(`!!raw-loader!./${svg}.svg`).default;
```
Comment (ixrock) [inline src/renderer/components/icon/icon.tsx:97]: How does it work / what's fixing? 
There are some working usages of `svg` prop in codebase:

image

File: src/renderer/components/hotbar/hotbar-icon.tsx [hunk 2]
```diff
@@ -55,6 +56,21 @@ export class HotbarIcon extends React.Component<Props> {
...
+    if (!category) {
+      return <Icon material="bug_report" className={className} />;
+    }
+
+    if (category.metadata.icon.includes("<svg")) {
+      return <Icon svg={category.metadata.icon} className={className} />;
+    } else {
+      return <Icon material={category.metadata.icon} className={className} />;
+    }
+  }
```
Comment (ixrock) [inline src/renderer/components/hotbar/hotbar-icon.tsx:72]: **nit**: I'd better to use `const iconProps: IconProps = {};` with `if-else` updates and `return <Icon {...iconProps}/>`.

**Review by ixrock — APPROVED** [review]

File: src/renderer/components/icon/icon.tsx (hunk 1 above)
Comment (jakolehm) [reply src/renderer/components/icon/icon.tsx:97]: Those usages expect that `svg` resides within `components/icon` folder. This PR allows to pass raw svg string to `Icon` component.

**Pull request merged**